tests/lib.sh

```diff
@@ -36,12 +39,15 @@ check_signature () {
   rlen=$((${#r_val}/2))
   slen=$((${#s_val}/2))
 
-  rfmt="02$(printf "%x" $rlen)${r_val}"
-  sfmt="02$(printf "%x" $slen)${s_val}"
+  rfmt="02 $(printf "%x" $rlen) ${r_val}"
+  echo $rfmt > RFMT
+  sfmt="02 $(printf "%x" $slen) ${s_val}"
+  echo $sfmt > SFMT
 
-  SIG="30$(printf "%x" $(($rlen+$slen+4)))$rfmt$sfmt"
+  SIG="30 $(printf "%x" $(($rlen+$slen+4))) $rfmt $sfmt"
+  echo $SIG > sigdump
 
-  xxd -r -ps <<<"$1" | blake2b_lock_hash | xxd -p -r | openssl pkeyutl -verify -pubin -inkey tests/public_key_0_0.pem -sigfile <(xxd -r -ps <<<"$SIG")
+  xxd -r -ps <<<"$1" | tee dumpfile0 | blake2b_p | tee hashfile | xxd -p -r | openssl pkeyutl -verify -pubin -inkey tests/public_key_0_0_0.pem -sigfile <(xxd -r -ps <<<"$SIG")
```

Comment:
is public_key_0_0_0.pem different from public_key_0_0.pem?

Reply:
Yes. The 0s are the derivation path after the 44'/309' part.

Reply:
ok, that file needs to be checked in as well i think